File: selectionsort.c
#include <stdio.h>

void swap(int* a, int* b) {
    int t = *a;
    *a = *b;
    *b = t;
}
void selsort(int arr[], int n) {
    int i, j, k;
    for (i = 0; i < n - 1; i++) {
        k = i;
        for (j = i + 1; j < n; j++)
            if (arr[j] < arr[k])
                k = j;
        swap(&arr[k], &arr[i]);
    }
}
void array(int arr[], int size) {
    int i;
    for (i=0; i < size; i++)
        printf("%d ", arr[i]);
    printf("\n");
}
int main() {
    int arr[] = {10,5,6,2,3,7};
    int n = sizeof(arr)/sizeof(arr[0]);
    selsort(arr, n);
    printf("Sorted array: \n");
    array(arr, n);
    return 0;
}

File: selectionsort.py
def swap(arr, i, j):
    arr[i], arr[j] = arr[j], arr[i]

def selsort(arr):
    n = len(arr)
    for i in range(n - 1):
        k = i
        for j in range(i + 1, n):
            if arr[j] < arr[k]:
                k = j
        swap(arr, i, k)

def array(arr):
    print(" ".join(map(str, arr)))

arr = [10, 5, 6, 2, 3, 7]
selsort(arr)
print("Sorted array:")
array(arr)